services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/SegmentServiceImpl.java

```diff
@@ -104,43 +107,46 @@ public Long addVlanEntity(String vlanId, String networkType, String vpcId) throw
      */
     @Override
     @DurationStatistics
-    public Long addVxlanEntity(String vxlanId, String networkType, String vpcId) throws Exception {
+    public Long addVxlanEntity(String vxlanId, String networkType, String vpcId, Integer mtu) throws Exception {
 
         Long key = null;
-        String rangeId = null;
+        String partitionStringFormat = null;
         Random ran = new Random();
-        Map<Integer, String> partitionsAndRangeIds = new HashMap<>();
+//        Map<Integer, String> partitionsAndRangeIds = new HashMap<>();
 
         try {
             NetworkVxlanType vxlan = new NetworkVxlanType();
+            // the number of randomly allocating partition in vxlan range
+            int round = 1;
+
+            while (round <= ConstantsConfig.MAX_ROUNDS) {
+                // Randomly allocate a partition and check if the partition exist in db
+                int partition = ran.nextInt(NetworkType.VXLAN_PARTITION);
+                partitionStringFormat = partition + "";
+                NetworkVxlanRange networkVxlanRange = this.vxlanRangeRepository.findItem(partitionStringFormat);
+                if (networkVxlanRange == null) {
+//                  rangeId = UUID.randomUUID().toString();
+                    int firstKey = partition * NetworkType.VXLAN_ONE_PARTITION_SIZE;
+                    int lastKey = (partition + 1) * NetworkType.VXLAN_ONE_PARTITION_SIZE;
+                    NetworkRangeRequest request = new NetworkRangeRequest(partitionStringFormat, networkType, partition, firstKey, lastKey);
+                    this.vxlanRangeRepository.createRange(request);
```

Comment:
Recommended to return id if a creation is successful.

Something like：
```
partitionStringFormat = this.vxlanRangeRepository.createRange(request);
```

Reply:
Sure